Tests/SourceKitDTests/CrashRecoveryTests.swift

```diff
@@ -45,7 +46,20 @@ final class CrashRecoveryTests: XCTestCase {
     let ws = try! staticSourceKitTibsWorkspace(name: "sourcekitdCrashRecovery")!
     let loc = ws.testLoc("loc")
 
+    // Open the document. Wait for the semantic diagnostics to know it has been fully opened and we are not entering any data races about outstanding diagnostics when we crash sourcekitd.
+
+    let documentOpened = self.expectation(description: "documentOpened")
+    documentOpened.expectedFulfillmentCount = 2
+    ws.sk.handleNextNotification({ (note: LanguageServerProtocol.Notification<PublishDiagnosticsNotification>) in
+      log("Received diagnostics for open - syntactic")
+      documentOpened.fulfill()
+    })
+    ws.sk.appendOneShotNotificationHandler({ (note: LanguageServerProtocol.Notification<PublishDiagnosticsNotification>) in
+      log("Received diagnostics for open - semantic")
+      documentOpened.fulfill()
+    })
     try! ws.openDocument(loc.url, language: .swift)
+    self.wait(for: [documentOpened], timeout: 5)
```

Comment:
We should use a larger base timeout for an AST build since CI tends to be slower than we expect sometimes.  Otherwise LGTM.

Reply:
Increased the timeout to 10 seconds. Or do you think we need more?